src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -627,10 +627,22 @@ public function walkNumComparisonConstraint(QOM\PropertyValueInterface $property
         $alias = $this->getTableAlias($propertyOperand->getSelectorName() . '.' . $propertyOperand->getPropertyName());
         $property = $propertyOperand->getPropertyName();
 
-        return
-            $this->sqlXpathExtractNumValue($alias, $property) . " " .
-            $operator . " " .
-            $literalOperand->getLiteralValue();
+
+        if ($this->platform instanceof MySqlPlatform && $operator == '=') {
```

Comment:
``if ($this->platform instanceof MySqlPlatform && $operator == '=') {``

=>

``if ($this->platform instanceof MySqlPlatform && '=' === $operator) {``